User sign up

shernanm opened the pull request:
## Summary

* The registration of users and persistence in database was implemented, with the fields:
  * First name.
  * Last name.
  * E-mail.
  * Password.

## Trello Card

https://trello.com/c/JPNH1VeU

Commit:
API consumption of albums is added

Commit:
Added code for a bad request

Commit:
Adding user model

Commit:
Changes requested to API consumption

Commit:
Merge branch 'api-integration' of https://github.com/wolox-training/shm-express-js into user-model

Commit:
wip

Commit:
merge with master

Commit:
Adjustments were made to the model and migration

Commit:
wip

Commit:
Changes are made in the definition of the model and migration.

Commit:
The deleted_at property is added to the migration and configurations are made in the model.

Commit:
Merge branch 'user-model' into user-sign-up

Commit:
wip

Commit:
The deleted_at property is modified

Commit:
Merge branch 'user-model' into user-sign-up

Commit:
The user registry was implemented.

shernanm committed:
Merge branch 'master' into user-sign-up

molinajulian commented on app/services/users.js:
```diff
@@ -0,0 +1,9 @@
+exports.userRegister = user =>
+  users.create(user).catch(err => {
+    logger.info(`Error trying to create the user ${user.firstName} ${user.lastName}`);
+    return Promise.reject(errors.databaseError(`${err.errors[0].message}`));
```
If throw error, you don't need Promise.reject

molinajulian replied:
Use a generic message when db fail

molinajulian commented on app/util.js:
```diff
@@ -0,0 +1,23 @@
+exports.paramsValidation = (pass, email) => {
+  const regexPass = /^([a-z0-9]){8,}$/i;
+  const regexMail = /^[a-z0-9._-]+@wolox.(co|cl|com|ar|com.ar)+$/i;
+  if (!regexPass.test(pass) || !regexMail.test(email)) {
```
Use ternary operator for returns

molinajulian commented on app/util.js:
```diff
@@ -0,0 +1,23 @@
+const errors = require('./errors');
+
+exports.paramsValidation = (pass, email) => {
+  const regexPass = /^([a-z0-9]){8,}$/i;
```
Great! Fantastic shape for resolve this

molinajulian commented on app/util.js:
```diff
@@ -0,0 +1,23 @@
+
+exports.passwordEncryption = user =>
+  bcrypt
+    .hash(user.password, 10)
```
Use env var for second parameter and default value is 10

molinajulian commented on app/util.js:
```diff
@@ -0,0 +1,23 @@
+      password
+    }))
+    .catch(err =>
+      Promise.reject(errors.encryptionError(`Error trying to encrypt the password. ${err.message}`))
```
Same with other comment for errors. Throw errors directly

molinajulian commented on app/controllers/user.js:
```diff
@@ -0,0 +1,23 @@
+const logger = require('../logger');
+
+exports.createUser = (req, res, next) => {
+  const user = req.body;
```
Deconstruct req for body :+1:

molinajulian commented on app/controllers/user.js:
```diff
@@ -0,0 +1,23 @@
+      return res.status(201).send({
+        firstName: user.firstName,
+        lastName: user.lastName,
+        message: `User has been created successfully with ID ${response.id}`
```
Delete this message

Commit:
Changes requested for user sign up
Changes were made in the validation of the regular expressions, in the capture of the errors and an environment variable was created for the saltRounds used in the encryption of the passwords.

Commit:
Merge branch 'user-sign-up' of https://github.com/wolox-training/shm-express-js into user-sign-up

raparicio6 commented on app/controllers/user.js:
```diff
@@ -0,0 +1,22 @@
+const logger = require('../logger');
+
+exports.createUser = (req, res, next) => {
+  const { body } = req;
```
Apply destructuring with the things inside req.body.
We use destructuring in order to avoid making object.property1, object.property2, etc many times

raparicio6 commented on app/controllers/user.js:
```diff
@@ -0,0 +1,22 @@
+  }
+  return validations
+    .passwordEncryption(body)
+    .then(userService.userRegister)
```
Nice

raparicio6 commented on app/services/users.js:
```diff
@@ -0,0 +1,9 @@
+const { users } = require('../models');
```
We use models in upper case and in singular

Commit:
Modifications made to the controller, service and user model.
